packages/embarkjs/src/blockchain.js

```diff
@@ -135,16 +154,16 @@ Blockchain.doConnect = function(connectionList, opts, doneCb) {
       } else {
         connectHttp(connectionString, next);
       }
-    }, function(_err, result) {
+    }, async function(_err, result) {
       if (!result.connected || result.error) {
         const connectionError = new BlockchainConnectionError(connectionErrs);
-        cb(connectionError);
+        await cb(connectionError);
```

Comment:
awaiting a `cb` is pretty weird. Why do we have to do that?

Reply:
This `cb` [corresponds](https://github.com/embark-framework/embark/blob/refactor/promises/packages/embarkjs/src/blockchain.js#L49-L53) (callback of `todo`) to the performance of zero to any number of `execWhenReady` tasks, any of which could be async. Previously, those tasks could possibly be ongoing while other things are happening at the same time, e.g. `doneCb`.

Also, I agree 100%, it's weird and I don't like it, but the whole `Blockchain.connect` and friends are a complex beast.  I'm open to suggestions on how we can improve it all.

Reply:
Yeah, the connect workflow might need refactoring at some point. We can leave it like this for now.